frontend/public/lib/socialMediaOperations.js

```diff
@@ -0,0 +1,138 @@
+import InstagramIcon from "@material-ui/icons/Instagram";
+import TwitterIcon from "@material-ui/icons/Twitter";
+import YouTubeIcon from "@material-ui/icons/YouTube";
+import FacebookIcon from "@material-ui/icons/Facebook";
+import LinkedInIcon from "@material-ui/icons/LinkedIn";
+
+import { apiRequest } from "./apiOperations";
+
+export function verifySocialMediaLink(socialChannel, url, texts) {
+  let error = "";
+  // matches http://, https:// , https://www. ,http://www.
+  const regexPrefix = "^(http)(?:s)?(://)(?:www.)?";
+  // matches.com/ anything
+  const regexSuffix = ".com/.+$";
+  let regex;
+  let matches = false;
+
+  switch (socialChannel.name) {
+    case "Twitter": // twitter
+      regex = new RegExp(regexPrefix + "twitter" + regexSuffix);
```

Comment:
instead of hardcoded options we should use the base url here. Otherwise this will break if we add a new social (e.g. TikTok) in the future. It should be possible to add a new social without making changes to the code

Reply:
Using baseUrl instead of hardcoded string.